src/librustc_mir/interpret/snapshot.rs

```diff
@@ -77,7 +77,7 @@ impl<'mir, 'tcx> InfiniteLoopDetector<'mir, 'tcx> {
         }
 
         // Second cycle
-        Err(InterpError::InfiniteLoop.into())
+        Err(InterpError::ResourceExhaustion(ResourceExhaustionInfo::InfiniteLoop).into())
```

Comment:
use macros

Reply:
We do not have an macro for `ResourceExhaustion` yet. Should we create one?

Reply:
Seems reasonable to have `throw_exhaust` and `err_exhaust`, for consistency.